core/systems/threading/shared_thread.gd

```diff
@@ -101,6 +101,28 @@ func scheduled_exec(method: Callable, wait_time_ms: int) -> void:
 	mutex.unlock()
 
 
+## Cancels a given Sheduled Task before it is executed.
+func cancel_scheduled_exec(task: ScheduledTask) -> void:
+	mutex.lock()
+	if task not in scheduled_funcs:
+		logger.warn("Scheduled Task " + task.method.get_method() + " canceled but not found in scheduled functions."  )
+		return
```

Comment:
You need to `mutex.unlock()` before you return here, otherwise this mutex will be locked forever, resulting in a deadlock.